subworkflows/local/map_alignment.nf

```diff
@@ -12,58 +9,40 @@ include {
 
 workflow MAP_ALIGNMENT {
     take:
-        indexed_hits
-        mzml_files
+        runs_to_be_aligned
+        mzml
 
     main:
         ch_versions = Channel.empty()
-        // Calculate fdr for id based alignment
-        OPENMS_FALSEDISCOVERYRATE(indexed_hits)
-        ch_versions = ch_versions.mix(OPENMS_FALSEDISCOVERYRATE.out.versions.first().ifEmpty(null))
-        // Filter fdr for id based alignment
-        OPENMS_IDFILTER_FOR_ALIGNMENT(OPENMS_FALSEDISCOVERYRATE.out.idxml
-                                        .flatMap { it -> [tuple(it[0], it[1], null)]})
-        ch_versions = ch_versions.mix(OPENMS_IDFILTER_FOR_ALIGNMENT.out.versions.first().ifEmpty(null))
-
-        // Group samples together if they are replicates
-        ch_grouped_fdr_filtered = OPENMS_IDFILTER_FOR_ALIGNMENT.out.idxml
-            .map {
-                meta, raw ->
-                    [[id:meta.sample + "_" + meta.condition, sample:meta.sample, condition:meta.condition, ext:meta.ext], raw]
-                }
-            .groupTuple(by: [0])
-        // Compute alignment rt transformation
-        OPENMS_MAPALIGNERIDENTIFICATION(ch_grouped_fdr_filtered)
+        // Compute group-wise alignment rt transformation
+        OPENMS_MAPALIGNERIDENTIFICATION( runs_to_be_aligned )
         ch_versions = ch_versions.mix(OPENMS_MAPALIGNERIDENTIFICATION.out.versions.first().ifEmpty(null))
-        // Obtain the unique files that were present for the combined data
-        joined_trafos = OPENMS_MAPALIGNERIDENTIFICATION.out.trafoxml
+        // MapAligner module might return trafoXML files and meta information in the wrong order, so we need to take care of that
+        OPENMS_MAPALIGNERIDENTIFICATION.out.trafoxml
             .transpose()
             .flatMap {
                 meta, trafoxml ->
-                    ident = trafoxml.baseName.split('_-_')[0]
+                    ident = trafoxml.baseName.split('_').first()
```

Comment:
Is there a difference in result if you run this with and without filtering? The pipeline works with a lot of underscores just making sure that the right information is passed further

Reply:
What do you mean by filtering here? The mapalginer determines the shift of RT for each peptide hit 🤔 
I tried to give each tool an short and descriptive output name. I will double check the orientations again though

Reply:
Double checked ✅